matching-notification-rest/src/main/java/org/phenotips/matchingnotification/rest/MatchingNotificationResource.java

```diff
@@ -93,6 +110,31 @@ Response getMatches(@FormParam("minScore") @DefaultValue("0.5") double minScore,
         @FormParam("fromDate") @DefaultValue("") String fromDate,
         @FormParam("toDate") @DefaultValue("") String toDate);
 
+    /**
+     * Returns a JSON object containing all matches for a provided {@code reference patient}, filtered by
+     * parameters. The following additional parameters may be specified:
+     *
+     * @param patientId the patient identifier whose matches we want to load
+     * @param minScore only matches with general score higher or equal to this value are returned
+     * @param minPhenScore only matches with phenotypic score higher or equal to this value are returned
+     * @param minGenScore only matches with genotypic score higher or equal to this value are returned
+     * @param fromDate if passed a date in the {@code yyyy-MM-dd} format, then only matches found on or after this date
+     *        will be returned; if {@code null} or an empty string, then no lower limit on the match date is considered
+     * @param toDate if passed a date in the {@code yyyy-MM-dd} format, then only matches found on or before this date
+     *        will be returned; if {@code null} or an empty string, then no upper limit on the match date is considered
+     * @return a response containing a JSON object with a list of matches
+     */
+    @GET
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/patients/{patientId}")
+    Response getMatchesForPatient(@PathParam("patientId") String patientId,
```

Comment:
I think we decided that for a single patient we don't need all those parameters, we just load all available matches for the patient. I can see an argument for keeping `fromDate` and `toDate` (when used via scripts to load only new matches), but for a single patient doing extra filtering on the back-end doesn't make much sense I think. We also talked about removing the score fields form the UI, so it will be impossible to change it as well. @marta- ?

Reply:
IMO It's OK to have all parameters for when patient id is present, in case we want to do some sort of filtering via API later. I don't see a problem with keeping them all.